researcher.lic

```diff
@@ -0,0 +1,178 @@
+      check_research
+      loop do
+        pause 5
+        check_research
+        break unless researching
+      end
+    else
+      echo "Using YAML settings is not supported at the moment, please specify a research target, or type ';researcher help' for a list of targets"
+      exit
+    end
+  end
+
+  def convert_guild_skills(skill)
+    case skill
+      when 'astro', 'astrology', 'plane'
+        'plane'
+      when 'summon', 'summoning', 'planes'
+        'planes'
+      when 'theurgy', 'road'
+        'road'
+    end 
+  end
+
+
+  def set_researching(status)
+    UserVars.researcher['researching'] = status
+    UserVars.researcher['timestamp'] = Time.now
+  end
+
+  def researching
+    return false if Flags['research-partial']
+    return false if Flags['research-complete']
+    return false unless DRSpells.active_spells.include?('Gauge Flow')
+    UserVars.researcher['researching']
+  end
+
+  def add_flags
+    # Use the same flags from crossing-training
+    Flags.add('research-partial', 'there is still more to learn before you arrive at a breakthrough', 'distracted by combat', 'distracted by your spellcasting', 'You lose your focus on your research project', 'you forget what you were') unless Flags['research-partial']
+    Flags.add('research-complete', '^Breakthrough!', 'You complete reviewing your knowledge of the \w+ symbiosis and commit the details to memory') unless Flags['research-complete']
+  end
+
+  def check_status
+    case DRC.bput('research status', 'You estimate that you will complete it a few minutes from now', "You're not researching anything", "You have completed .+ of a project about")
+      when /You estimate that you will complete it a few minutes from now/
+        set_researching(true)
+      when /You're not researching anything/, /You have completed .+ of a project about/
+        set_researching(false)
+    end
+  end
+
+  def check_research
+    if Flags['research-partial']
+      Flags.reset('research-partial')
+      set_researching(false)
+      start_research
+    elsif Flags['research-complete']
+      Flags.reset('research-complete')
+      set_researching(false)
+      @current_topic = nil
+    else
+      start_research
+    end
+  end
+
+  def start_research
+    return if researching
+
+    research_start_msgs = [
+      'You focus', 
+      'You tentatively', 
+      'You confidently', 
+      'Abandoning the normal', 
+      'You are already busy',
+      'You start to research',
+      'With a mixture of rational concern',
+      'You expertly coach',
+      'You begin to bend',
+      'You bring yourself',
+      'Inwardly, you make an exercise',
+      'You orient in the figurative direction'
+    ]
+
+    research_start_fail_msgs = [
+      'You cannot begin',
+      'Usage:', 
+      'You do not know how to research'
+    ]
+
+
+    check_gaf
+    @current_topic = 'stream' if @current_topic == 'attunement'
```

Comment:
I suggest moving this up with line 22 where `@current_topic` is assigned (then it can be treated as read-only for the rest of the script).

Reply:
Sounds fine to me. Done